### PR TITLE
Ranker1p4

### PR DESCRIPTION
**Pull request by raynCovar**
Updates to ranker to handle trapi 1.4 messages.

### CODE REVIEW

File: ranker/shared/ranker_obj.py [hunk 1]
```diff
@@ -77,55 +77,56 @@ def rank(self, answers, jaccard_like=False):
         for answer in answers:
             answers_.append(self.score(answer, jaccard_like=jaccard_like))
 
-        answers.sort(key=itemgetter('score'), reverse=True)
```
[the remark this answers is not in the document]
Comment (raynCovar) [reply ranker/shared/ranker_obj.py:80]: I kinda figured the UI would sort. If not, max over analyses is probably the right answer.

**Comment by cbizon** [comment]
What does "auxilliary graphs are not fully considered" mean?

File: ranker/shared/ranker_obj.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbizon) [reply ranker/shared/ranker_obj.py:80]: I would still like these to be sorted, max over analysis makes sense to me.

**Comment by raynCovar** [comment]
> 

We don't currently have any responses to work off of so were planning to add it in after we can get a number of example responses

**Comment by cbizon** [comment]
Yes, but we do have a spec for responses, right?

**Comment by raynCovar** [comment]
Yep, I'll finish it up - I am expecting some bugs when we do get them

File: ranker/shared/ranker_obj.py [hunk 2]
```diff
@@ -190,6 +190,14 @@ def get_rgraph(self, answer):
         """Get "ranker" subgraph."""
         rnodes = set()
         redges = []
+        dummy_ind = 0
+        #adds edges and nodes from support graphs into the answer edge bindings and node bindings
+        for i_analysis in range(len(answer.get('analyses',[]))):
+            for sg in answer['analyses'][i_analysis]['support_graphs']:
```
Comment (maximusunc) [inline ranker/shared/ranker_obj.py:196]: You should `.get('support_graphs', [])` because they aren't guaranteed to exist.

**Review by maximusunc — APPROVED** [review]
Looks good as far as I can tell!

**Review by uhbrar — APPROVED** [review]
This looks good to me now.

**Review by cbizon — APPROVED** [review]

**Pull request merged**